diff --git a/inference/CMakeLists.txt b/inference/CMakeLists.txt
index 329e07fe..7ac41869 100644
--- a/inference/CMakeLists.txt
+++ b/inference/CMakeLists.txt
@@ -105,7 +105,7 @@ endif()
 
 # Documentation: https://cliutils.gitlab.io/modern-cmake/chapters/projects/submodule.html
 # Ensures the submodules are set correctly during a build.
-find_package(Git QUIET)
+#find_package(Git QUIET)
 if(GIT_FOUND AND EXISTS "${REPOSITORY_ROOT_DIR}/.git")
 # Update submodules as needed
     option(GIT_SUBMODULE "Check submodules during build" ON)
diff --git a/inference/marian-fork/src/CMakeLists.txt b/inference/marian-fork/src/CMakeLists.txt
index b303d289..ea8a9bf1 100644
--- a/inference/marian-fork/src/CMakeLists.txt
+++ b/inference/marian-fork/src/CMakeLists.txt
@@ -128,7 +128,7 @@ if (NOT USE_WASM_COMPATIBLE_SOURCE)
   endif(USE_FBGEMM)
 endif()
 
-add_library(marian STATIC ${MARIAN_SOURCES})
+add_library(marian SHARED ${MARIAN_SOURCES})
 
 target_compile_options(marian PRIVATE ${ALL_WARNINGS})
 
diff --git a/inference/src/translator/CMakeLists.txt b/inference/src/translator/CMakeLists.txt
index df8d8145..19876f5d 100644
--- a/inference/src/translator/CMakeLists.txt
+++ b/inference/src/translator/CMakeLists.txt
@@ -2,7 +2,7 @@
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/project_version.h.in
                ${CMAKE_CURRENT_BINARY_DIR}/project_version.h @ONLY)
 
-add_library(bergamot-translator-source STATIC
+set(BERGAMOT_SOURCES
     byte_array_util.cpp
     text_processor.cpp
     translation_model.cpp 
@@ -19,6 +19,9 @@ add_library(bergamot-translator-source STATIC
     html.cpp
     xh_scanner.cpp
 )
+
+add_library(bergamot-translator-source SHARED ${BERGAMOT_SOURCES})
+
 if (USE_WASM_COMPATIBLE_SOURCE)
   # Using wasm compatible sources should include this compile definition;
   # Has to be done here because we are including marian headers + some sources
