diff --git a/inference/CMakeLists.txt b/inference/CMakeLists.txt
index 329e07fe..7ac41869 100644
--- a/inference/CMakeLists.txt
+++ b/inference/CMakeLists.txt
@@ -105,7 +105,7 @@ endif()
 
 # Documentation: https://cliutils.gitlab.io/modern-cmake/chapters/projects/submodule.html
 # Ensures the submodules are set correctly during a build.
-find_package(Git QUIET)
+#find_package(Git QUIET)
 if(GIT_FOUND AND EXISTS "${REPOSITORY_ROOT_DIR}/.git")
 # Update submodules as needed
     option(GIT_SUBMODULE "Check submodules during build" ON)
