diff --git a/inference/marian-fork/src/CMakeLists.txt b/inference/marian-fork/src/CMakeLists.txt
index b303d289..70726e3a 100644
--- a/inference/marian-fork/src/CMakeLists.txt
+++ b/inference/marian-fork/src/CMakeLists.txt
@@ -128,7 +128,9 @@ if (NOT USE_WASM_COMPATIBLE_SOURCE)
   endif(USE_FBGEMM)
 endif()
 
-add_library(marian STATIC ${MARIAN_SOURCES})
+#add_library(marian STATIC ${MARIAN_SOURCES})
+
+add_library(marian SHARED ${MARIAN_SOURCES})
 
 target_compile_options(marian PRIVATE ${ALL_WARNINGS})
 
diff --git a/inference/src/translator/CMakeLists.txt b/inference/src/translator/CMakeLists.txt
index df8d8145..425b7391 100644
--- a/inference/src/translator/CMakeLists.txt
+++ b/inference/src/translator/CMakeLists.txt
@@ -2,7 +2,8 @@
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/project_version.h.in
                ${CMAKE_CURRENT_BINARY_DIR}/project_version.h @ONLY)
 
-add_library(bergamot-translator-source STATIC
+#add_library(bergamot-translator-source STATIC
+set(BERGAMOT_SOURCES
     byte_array_util.cpp
     text_processor.cpp
     translation_model.cpp 
@@ -19,6 +20,9 @@ add_library(bergamot-translator-source STATIC
     html.cpp
     xh_scanner.cpp
 )
+
+
+add_library(bergamot-translator-source SHARED ${BERGAMOT_SOURCES})
 if (USE_WASM_COMPATIBLE_SOURCE)
   # Using wasm compatible sources should include this compile definition;
   # Has to be done here because we are including marian headers + some sources
