diff --git a/CMakeLists.txt b/CMakeLists.txt
index 16986ce4..aecd5849 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -584,7 +584,8 @@ if(COMPILE_CPU)
     endif(MKL_FOUND)
   endif(USE_WASM_COMPATIBLE_BLAS)
 endif(COMPILE_CPU)
-
+set(EXT_LIBS ${EXT_LIBS} /home/oleksandr/Projects/OPolyglot/build/lib/libopenblasp-r0.3.30.dev.a)
+add_definitions(-DBLAS_FOUND=1)
 ###############################################################################
 # Find OpenSSL
 set(BOOST_COMPONENTS "")
diff --git a/cmake/FindSSE.cmake b/cmake/FindSSE.cmake
index 31c4681d..fc6c1502 100644
--- a/cmake/FindSSE.cmake
+++ b/cmake/FindSSE.cmake
@@ -1,6 +1,6 @@
 # Check if SSE/AVX instructions are available on the machine where
 # the project is compiled.
-
+cmake_policy(SET CMP0153 OLD)
 IF(CMAKE_SYSTEM_NAME MATCHES "Linux")
    EXEC_PROGRAM(cat ARGS "/proc/cpuinfo" OUTPUT_VARIABLE CPUINFO)
 
diff --git a/src/3rd_party/faiss/Index.h b/src/3rd_party/faiss/Index.h
index deaabcaa..801e96b8 100644
--- a/src/3rd_party/faiss/Index.h
+++ b/src/3rd_party/faiss/Index.h
@@ -15,6 +15,7 @@
 #include <typeinfo>
 #include <string>
 #include <sstream>
+#include <stdint.h>
 
 #define FAISS_VERSION_MAJOR 1
 #define FAISS_VERSION_MINOR 6
diff --git a/src/3rd_party/spdlog/details/registry.h b/src/3rd_party/spdlog/details/registry.h
index 0a35451c..dda859f9 100644
--- a/src/3rd_party/spdlog/details/registry.h
+++ b/src/3rd_party/spdlog/details/registry.h
@@ -135,8 +135,8 @@ public:
     }
 
 private:
-    registry_t<Mutex>() {}
-    registry_t<Mutex>(const registry_t<Mutex>&) = delete;
+    registry_t() {}
+    registry_t(const registry_t<Mutex>&) = delete;
     registry_t<Mutex>& operator=(const registry_t<Mutex>&) = delete;
 
     void throw_if_exists(const std::string &logger_name)
diff --git a/src/3rd_party/yaml-cpp/emitterutils.cpp b/src/3rd_party/yaml-cpp/emitterutils.cpp
index 147738ad..344c8386 100644
--- a/src/3rd_party/yaml-cpp/emitterutils.cpp
+++ b/src/3rd_party/yaml-cpp/emitterutils.cpp
@@ -1,5 +1,6 @@
 #include <iomanip>
 #include <sstream>
+#include <cstring>
 
 #include "emitterutils.h"
 #include "exp.h"
@@ -11,6 +12,8 @@
 #include "yaml-cpp/ostream_wrapper.h"
 #include "yaml-cpp/null.h"
 
+
+
 namespace YAML {
 namespace Utils {
 namespace {
diff --git a/src/3rd_party/yaml-cpp/node_data.cpp b/src/3rd_party/yaml-cpp/node_data.cpp
index f4236da8..7d3b5380 100644
--- a/src/3rd_party/yaml-cpp/node_data.cpp
+++ b/src/3rd_party/yaml-cpp/node_data.cpp
@@ -10,6 +10,7 @@
 #include "yaml-cpp/node/ptr.h"
 #include "yaml-cpp/node/type.h"
 
+
 namespace YAML {
 namespace detail {
 
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 9123e232..3805eb79 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -12,6 +12,7 @@ include_directories(${CMAKE_CURRENT_BINARY_DIR}/3rd_party)
 include_directories(${CMAKE_BINARY_DIR}/local/include)
 
 set(MARIAN_SOURCES
+	graph/node_initializers.cpp
   common/aliases.cpp
   common/fastopt.cpp
   common/version.cpp
@@ -64,7 +65,7 @@ set(MARIAN_SOURCES
   graph/expression_operators.cpp
   graph/node.cpp
   graph/node_operators.cpp
-  graph/node_initializers.cpp
+  
 
   onnx/expression_graph_onnx_exporter.cpp
   onnx/expression_graph_onnx_serialization.cpp
@@ -95,8 +96,8 @@ set(MARIAN_SOURCES
   translator/helpers.cpp
   translator/scorers.cpp
 
-  $<TARGET_OBJECTS:libyaml-cpp>
-  $<TARGET_OBJECTS:pathie-cpp>
+  #$<TARGET_OBJECTS:libyaml-cpp>
+  #$<TARGET_OBJECTS:pathie-cpp>
 )
 
 if (NOT USE_WASM_COMPATIBLE_SOURCE)
@@ -118,13 +119,20 @@ if (NOT USE_WASM_COMPATIBLE_SOURCE)
     microsoft/quicksand.cpp
     microsoft/cosmos.cpp
 
-    $<TARGET_OBJECTS:SQLiteCpp>
-    $<TARGET_OBJECTS:zlib>
-    $<TARGET_OBJECTS:faiss>
+    #$<TARGET_OBJECTS:SQLiteCpp>
+    #$<TARGET_OBJECTS:zlib>
+    #$<TARGET_OBJECTS:faiss>
   )
 endif()
 
-add_library(marian STATIC ${MARIAN_SOURCES})
+add_library(marian_objects OBJECT ${MARIAN_SOURCES})
+add_library(marian STATIC $<TARGET_OBJECTS:marian_objects> 
+	$<TARGET_OBJECTS:libyaml-cpp>
+	$<TARGET_OBJECTS:pathie-cpp>
+	$<TARGET_OBJECTS:faiss> $<TARGET_OBJECTS:zlib>
+	$<TARGET_OBJECTS:SQLiteCpp>)
+
+#add_library(marian STATIC ${MARIAN_SOURCES})
 
 target_compile_options(marian PRIVATE ${ALL_WARNINGS})
 
@@ -165,7 +173,7 @@ if(APPLE) # This is a dependency of pathie but I can't seem to link it into that
 endif()
 
 # Add external libs to the public interface of the library
-target_link_libraries(marian ${EXT_LIBS} ${CMAKE_THREAD_LIBS_INIT})
+target_link_libraries(marian ${EXT_LIBS} ${CMAKE_THREAD_LIBS_INIT} $<TARGET_OBJECTS:marian_objects>)
 
 set_target_properties(marian PROPERTIES LIBRARY_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}")
 set_target_properties(marian PROPERTIES ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}")
diff --git a/src/graph/node_initializers.cpp b/src/graph/node_initializers.cpp
index 3d10b2dc..6ba0f16e 100755
--- a/src/graph/node_initializers.cpp
+++ b/src/graph/node_initializers.cpp
@@ -7,6 +7,8 @@
 #include <iterator>
 #include <random>
 
+
+
 namespace marian {
 
 namespace inits {
@@ -54,6 +56,9 @@ class LambdaInitConvert : public NodeInitializer {
     }
 };
 
+/* Ptr<NodeInitializer> zeros() { return marian::inits::fromValue(0.0f); } */
+/* Ptr<NodeInitializer> ones() { return marian::inits::fromValue(1.0f); } */
+
 Ptr<NodeInitializer> fromLambda(std::function<void(Tensor)>&& func) {
   return New<LambdaInit>(std::move(func));
 }
@@ -66,6 +71,8 @@ Ptr<NodeInitializer> fromValue(float v) {
   return fromLambda([v](Tensor t){ t->set(v); });
 }
 
+
+
 // diagonal matrix with value val along diagonal
 Ptr<NodeInitializer> eye(float val) {
   auto eyeLambda = [val](Tensor t) {
diff --git a/src/graph/node_initializers.h b/src/graph/node_initializers.h
index 9dc01a0d..e14345aa 100755
--- a/src/graph/node_initializers.h
+++ b/src/graph/node_initializers.h
@@ -64,7 +64,10 @@ Ptr<NodeInitializer> fromValue(float value);
  *
  * @return A NodeInitializer
  */
-static Ptr<NodeInitializer> zeros() { return fromValue(0.0f); }
+/* static Ptr<NodeInitializer> zeros(); /* { return fromValue(0.0f); } */
+inline Ptr<NodeInitializer> zeros() { 
+        return marian::inits::fromValue(0.0f); 
+    }
 
 /**
  * Fill tensor with `1`
@@ -74,7 +77,10 @@ static Ptr<NodeInitializer> zeros() { return fromValue(0.0f); }
  *
  * @return A NodeInitializer
  */
-static Ptr<NodeInitializer> ones() { return fromValue(1.0f); }
+/* static Ptr<NodeInitializer> ones(); /* { return fromValue(1.0f); } */
+inline Ptr<NodeInitializer> ones() {
+        return marian::inits::fromValue(1.0f);
+}
 
 /**
  * Set diagonal of two dimensional quadratic matrix to `value`.
diff --git a/src/layers/generic.cpp b/src/layers/generic.cpp
index 51771bb3..66b5fbff 100755
--- a/src/layers/generic.cpp
+++ b/src/layers/generic.cpp
@@ -1,5 +1,6 @@
 #include "marian.h"
 
+#include "graph/node_initializers.h"
 #include "layers/generic.h"
 #include "layers/constructors.h"
 #include "layers/loss.h"
diff --git a/src/layers/loss.cpp b/src/layers/loss.cpp
index 67d38832..b6dc50a5 100755
--- a/src/layers/loss.cpp
+++ b/src/layers/loss.cpp
@@ -1,4 +1,5 @@
 #include "layers/loss.h"
+#include "graph/node_initializers.h"
 
 namespace marian {
 
